### ISSUE TITLE
disapprove forEach

### ISSUE THREAD

**Issue by bergus** [airbnb/javascript#1068]
The [no-iterators rule](/airbnb/javascript#iterators--nope) seems to be well-reasoned, but the provided example entitles the `forEach` method as `// good`. This doesn't seem to fit together, all what `forEach` is about are side effects, it doesn't `return` anything. The example uses a mutable `sum` variable just like the `// bad` one. I don't see much difference (except for making it look less structured and more object-oriented). Is this really worth it?

I think that example should be entitled as `// bad` as well or deleted altogether. Instead an example for correct usage of `map` should be added.

**Comment by ljharb** [comment]
I'm fine with a better example using `map` - but using `forEach` is strictly MUCH better than a loop, if only because it's a gateway drug to using the rest of the iterator methods.

You'll note that the "good" forEach example is followed by a "best" reduce example - I'm not sure what an example of "map" would add.

**Comment by bergus** [comment]
A gateway drug, aha, well OK. I still wouldn't call it "good", maybe `// better` (than "bad") would be more appropriate.

As for a `map` example, why not just take one of the existing and write
```
// bad
const nums = [1, 2, 3];
const incs = [];
nums.forEach(x => {
  incs.push(x + 1);
});

// good
const nums = [1, 2, 3];
const incs = nums.map(x => x + 1);
```

**Comment by ljharb** [comment]
Adding a "map" example set, alongside the "sum" example set, seems great.

**Comment by ljharb** [comment]
Fixed by #1201.
